### DISCUSSION TITLE
Feature Request: Introduce NoEcho Attribute for GitHub Actions Inputs

### DISCUSSION THREAD

**Discussion by herzallah** [community/community#121196] [Actions]
### Select Topic Area

Product Feedback

### Body

**Overview**:
This proposal advocates for the introduction of a NoEcho attribute for inputs within the `workflow_dispatch` of GitHub Actions. This enhancement focuses on enhancing security by ensuring that sensitive data, such as passwords and API tokens, is masked in outputs, thereby protecting it from potential exposure.

**Feature Details:**
Implementing NoEcho would mask sensitive input values, substituting them with asterisks (*****) in all user interfaces, logs, and API responses where they might otherwise be displayed.

**Use Cases:**

Protecting sensitive credentials in settings where GitHub Actions logs might be accessible to multiple users.
Enhancing compliance with security standards in environments that require stringent data protection measures.
**Benefits:**
NoEcho would elevate the security features of GitHub Actions, aligning with best practices for data protection and mitigating the risk of sensitive data leaks in automation workflows.

**Implementation Suggestions:**

The NoEcho attribute should be included in the YAML configurations for workflow_dispatch inputs.
Backend adaptations to enforce this behaviour consistently across all GitHub Actions interfaces and outputs.

**Conclusion**:
Introducing the NoEcho attribute would significantly enhance the security mechanisms within GitHub Actions, providing users with crucial tools to maintain data confidentiality. 

We look forward to the consideration of this feature in an upcoming GitHub update.

**Comment by grandmas-favorite** [comment]
Hi @herzallah!

We appreciate your feedback and your valuable input. We want to be transparent about our current priorities. You can gain insights into the projects we’re currently focusing on for the upcoming year by checking our [public roadmap](https://github.com/github/roadmap/projects/1). While smaller fixes may not always be explicitly listed, they often get addressed when related work is in progress.

Your contributions are instrumental in making GitHub better for all users. Thank you for taking the time to share your thoughts with us, even if the particular request is not currently in active development. Your input is truly valuable and contributes to the continuous improvement of GitHub!
